sdk/aqueduct/table_artifact.py

```diff
@@ -321,6 +424,50 @@ def internal_std_metric(table: pd.DataFrame) -> float:
         )
         return self._apply_metric_to_table(internal_std_metric, metric_name, metric_description)
 
+    def _apply_check_to_table(
+        self,
+        check_function: Callable[..., bool],
+        check_name: str,
+        check_description: str,
+        severity: CheckSeverity = CheckSeverity.WARNING,
+    ) -> CheckArtifact:
+        zip_file = serialize_function(check_function)
+        function_spec = FunctionSpec(
+            type=FunctionType.FILE,
+            granularity=FunctionGranularity.TABLE,
+            file=zip_file,
+        )
+        op_spec = OperatorSpec(check=CheckSpec(level=severity, function=function_spec))
+
+        operator_id = generate_uuid()
+        output_artifact_id = generate_uuid()
+        apply_deltas_to_dag(
+            self._dag,
+            deltas=[
+                AddOrReplaceOperatorDelta(
+                    op=Operator(
+                        id=operator_id,
+                        name=check_name,
+                        description=check_description,
+                        spec=op_spec,
+                        inputs=[self._artifact_id],
+                        outputs=[output_artifact_id],
+                    ),
+                    output_artifacts=[
+                        aqueduct.artifact.Artifact(
+                            id=output_artifact_id,
+                            name=artifact_name_from_op_name(check_name),
+                            spec=ArtifactSpec(bool={}),
+                        )
+                    ],
+                ),
+            ],
+        )
+
+        return CheckArtifact(
+            api_client=self._api_client, dag=self._dag, artifact_id=output_artifact_id
+        )
+
```

Comment:
Isn't this method almost exactly the same as `_apply_metric_to_table()`? Can we avoid duplicating all this? 

From what I can tell, the only difference is in the OperatorSpec and ArtifactSpec. Can we do a similar thing as `wrap_spec()`, where that helper just takes in an OperatorSpec and figures out what to return? You can cast the return value to the appropriate artifact.